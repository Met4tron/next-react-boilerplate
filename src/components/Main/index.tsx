import * as C from './styles';
const Main = ({
  title = 'React Boilerplate',
  description = 'Typescript, ReactJS, NextJS e Styled Components'
}) => (
  <C.Wrapper>
    <C.Logo
      src="/img/logo.png"
      alt="Imagem contendo a palavra Boilerplate"
    ></C.Logo>
    <C.Title>{title}</C.Title>
    <C.Description>{description}</C.Description>
  </C.Wrapper>
);

export default Main;
