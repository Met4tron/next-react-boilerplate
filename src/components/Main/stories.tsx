import { Story, Meta } from '@storybook/react/types-6-0';
import Main from '.';

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Default',
    description: 'Default'
  }
} as Meta;

export const Basic: Story = (args) => <Main {...args} />;
