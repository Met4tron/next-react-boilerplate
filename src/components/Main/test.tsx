import { screen, render } from '@testing-library/react';

import Main from '.';

describe('<Main />', () => {
  it('should render the heading', () => {
    render(<Main />);

    expect(
      screen.getByRole('heading', { name: /react avancado/i })
    ).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<Main />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
