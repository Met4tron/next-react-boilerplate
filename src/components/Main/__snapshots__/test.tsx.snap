// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should match snapshot 1`] = `
.c0 {
  background-color: #06092b;
  color: #fff;
  width: 100%;
  height: 100%;
  padding: 3rem;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  width: 25rem;
  margin-bottom: 2rem;
}

.c2 {
  font-size: 2.5rem;
}

.c3 {
  font-size: 2rem;
  font-weight: 400;
}

<main
  class="c0"
>
  <img
    alt="Imagem contendo a palavra Boilerplate"
    class="c1"
    src="/img/logo.png"
  />
  <h1
    class="c2"
  >
    React Boilerplate
  </h1>
  <h2
    class="c3"
  >
    Typescript, ReactJS, NextJS e Styled Components
  </h2>
</main>
`;
